lib/charms/file_reverse_reader/file_reader.py

```diff
@@ -12,7 +12,7 @@ def readlines_reverse(qfile) -> Generator[int, None, None]:
     """Read the lines of a file in reverse order in a lazy way."""
     with qfile:
         qfile.seek(0, os.SEEK_END)
-        position = qfile.tell()
+        position = qfile.tell() -2
```

Comment:
A small comment to explain why `-2` ?

Reply:
the NGINX files print all the lines with a final \n, we need to ignore the last one to read the whole file backwards

Reply:
if not, the iterator will return an empty string (because the first read char is a \n), takewhile will stop the iteration and nothing will be yielded

Reply:
Can you add a small comment in the code ?